pkg/master/reconcilers/lease_test.go

```diff
@@ -417,24 +417,29 @@ func TestLeaseEndpointReconciler(t *testing.T) {
 	for _, test := range reconcileTests {
 		fakeLeases := newFakeLeases()
 		fakeLeases.SetKeys(test.endpointKeys)
-		registry := &registrytest.EndpointRegistry{
-			Endpoints: test.endpoints,
+		clientset := fake.NewSimpleClientset()
+		if test.endpoints != nil {
+			for _, ep := range test.endpoints.Items {
+				if _, err := clientset.CoreV1().Endpoints(ep.Namespace).Create(&ep); err != nil {
```

Comment:
This adds extra actions in the fake client.  Instead, build up the list of objects to pass to the `NewSimpleClientset`.  You may even be able to call test.endpoints directly as an arg depending on type.